package.json

```diff
@@ -152,6 +152,12 @@
             "description": "Output additional diagnostics to the Swift Output View.",
             "order": 8
           },
+          "swift.SDK": {
+            "type": "string",
+            "default": "",
+            "markdownDescription": "The path of the preferred Swift SDK.  The default is:\n- the result of `xcrun --sdk macosx --show-sdk-path` on macOS;\n- the value of `%SDKROOT%` on Windows.",
```

Comment:
```suggestion
            "markdownDescription": "The path of the Swift SDK.  The default is to use the SDK that came packaged with your swift installation",
```

Reply:
Oops, macOS SDKs never comes with Swift. It is always distributed standalone, and managed by `xcrun` by default. And on Linux we don’t quite use the concept of SDK, there’s also no "default SDK".

Reply:
I don't think this is resolved. If you want to add details about what the default is, it cannot be run this command to find out. It needs to be more descriptive. If you want to add more details, I have a settings documentation file in PR #240 which has more detail about each setting. It will need to be extended with the new settings

Reply:
I think this would definitely need extra documentation there. So change the description here to:

> The path of the SDK to compile against. The default SDK is determined by the environment on macOS and Windows.

Okay?

Reply:
@stevapple Need to mention Linux where it is packaged with the swift install

Reply:
On Linux no SDK will be set by default.

Setting an SDK also means to make it a sysroot for clang, which may cause troubles on Linux. We’re just including Swift libraries in Linux toolchain, and SDK means much more.

Reply:
Yes I understand but the text doesn't mention this. At least say this variable has no effect on Linux

Reply:
You can build a package with C/C++ (eg. `llbuild`) by `swift build -v` on each platform to verify this. On macOS and Windows, `-sdk` is set for `swiftc` and `-isysroot` is set for `clang`; on Linux neither will be set.

Reply:
How about

> The path of the SDK to compile against. The default SDK is determined by the environment on macOS and Windows. On Linux this setting will have no effect.

Reply:
Also wrong. On Linux explicitly-set SDK will still be respected.

No default ≠ No support

Reply:
Is setting the sdk on linux actually any use?

Reply:
Setting an SDK has two general purposes:
- Point to an SDK where SDK is necessary (macOS/Windows) and when the default SDK cannot be found, eg. when Xcode is missing on macOS;
- For cross-compilation.

On Linux it’s obviously the latter.